### PR TITLE
Automation/6809-chanel-automation-add-image-optimization-into-ci

### PR DESCRIPTION
**Pull request by IsraelleHub**
## Description of Change
We want to put some level of image file size optimization into the process to reduce the size of the files, which should help upload times and bundle size.

This compress can be added to each build, should be pretty lightweight.

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by IsraelleHub — COMMENTED** [review]
test

**Review by IsraelleHub — COMMENTED** [review]
test2

File: VAMobile/ios/fastlane/screenshots/ipadPro129-screen-1.png
Comment (narin) [inline VAMobile/ios/fastlane/screenshots/ipadPro129-screen-1.png:1]: All of these screenshots are new files and not modifications to the original. They should be in the `VAMobile/ios/fastlane/screenshots/en-US` folder. Looks like you missed the `en-US` in the image optimization output for iOS.

**Review by narin — CHANGES_REQUESTED** [review]

File: VAMobile/ios/fastlane/screenshots/ipadPro129-screen-1.png
Comment (IsraelleHub) [reply VAMobile/ios/fastlane/screenshots/ipadPro129-screen-1.png:1]: @narin should be good now. I haven't cleaned up the on-demand-build workflow. Waiting on final review to clean it.

Comment (narin) [reply VAMobile/ios/fastlane/screenshots/ipadPro129-screen-1.png:1]: I see the fix to the output folder, but all the newly added images are still present. Please remove all the newly added images in the `/screenshots` folder.

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:4]: Looks like these were copied over from the release workflow. Please rename workflow and update comment above describing the workflow.

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:29]: Are these list image steps before and after necessary? Maybe something more useful would be to list the # of images that were optimized afterwards. Could potentially count the # of files modified below on line 58.

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:36]: Consider a `cd VAMobile/android/fastlane/metadata/android/en-US/images` before running imagemin so that these lines are shorter and easier to read.

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:49]: `git add .` would include everything in the folder and `git add <folder>` is recursive so that makes most of the `git add`'s below necessary.

If we want to be certain we're commit **_only_** screenshots, the following 2 lines should be fine:

```
git add VAMobile/ios/fastlane/screenshots/en-US
git add VAMobile/android/fastlane/metadata/android/en-US/images
```

File: .github/workflows/optimize-images.yml
Comment (IsraelleHub) [reply .github/workflows/optimize-images.yml:4]: good catch

File: .github/workflows/optimize-images.yml
Comment (IsraelleHub) [reply .github/workflows/optimize-images.yml:29]: I added the list before and after for debug purposes. If it can count the number of opmitimized images, that will be great!!

**Comment by TKDickson** [comment]
Please update the name of the workflow to match the current pattern of `[Category] Workflow Name`. Looks like the categories we have so far are: Build, Code Quality, Documentation, e2e, Github Management, Release, and TestRail, so please pick whichever of those is relevant.

**Comment by IsraelleHub** [comment]
@TKDickson the workflow name right now is [build] optimize image. are you referring to this workflow or the required qa approval workflow

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:3]: Needs to be changed to describe this workflow

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (narin) [inline .github/workflows/pr-checks-for-qa-approvals.yml:17]: Is this change meant to be in this PR? If so, in the future please mention it in the description or leave a comment for the reviewer about it since it seems to be unrelated to the ticket.

**Review by narin — CHANGES_REQUESTED** [review]
Looking close. Some minor change requests and also there are 8 new images added to the `VAMobile/android/fastlane/metadata/android/en-US/images/` that shouldn't be in the PR.

File: .github/workflows/pr-checks-for-qa-approvals.yml
Comment (IsraelleHub) [reply .github/workflows/pr-checks-for-qa-approvals.yml:17]: Let me create PR for it.

**Comment by IsraelleHub** [comment]
@narin please review whenever you can!

File: .github/workflows/optimize-images.yml
Comment (narin) [inline .github/workflows/optimize-images.yml:2]: Please add space after `#` and `.`.

Otherwise this looks good.

**Review by narin — APPROVED** [review]

**Review by narin — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]

**Pull request merged**
